### PR TITLE
Adapt FIM integration tests for MacOS

### LINKED ISSUE #248: Make simple syscheck automated tests work for MacOS platform
| Working branch |
| --- |
|[dev-248-macos](https://github.com/wazuh/wazuh-qa/tree/dev-248-macos)|

## Description
Now that we have a decent set of integration tests for simple configurations, we should make them work for MacOS agents too. 

## Tasks
- [x] Prepare an environment to run syscheck in a MacOS agent
- [x] Assure only scheduled monitoring will be tested in a MacOS agent

Run all simple tests iteratively until the same results as Unix tests are obtained:
- [ ] ~test_audit~ (whodata is not available on MacOS)
- [x] test_basic_usage
- [ ] ~test_benchmark~
- [x] test_checks
- [x] test_ignore
- [x] test_invalid
- [x] test_nodiff
- [ ] ~test_prefilter_cmd~
- [x] test_recursion_level
- [x] test_report_changes
- [x] test_restrict
- [x] test_scan
- [ ] ~test_sync~ (No need to test this on MacOS as it's a Manager only functionality)
- [x] test_tags
- [x] test_ambiguous_configurations
- [x] test_skip (skip_nfs and skip_dev)

### PR DESCRIPTION
**Pull request by vicferpoy**
Hi team, this closes #248 .

This PR modifies `wazuh-testing` and some tests to make them work for MacOS agents.

In addition to this, it includes refactoring to some functions like `control_service` and some tests that were not refactored after the last changes.

I also added a new function to modify already created parameters from `ossec.conf` in all platforms, fixing the reconnect issue in agents and modified the conftest file to add debug parameters to our environment in every test (this had to be done manually).

## Tests performed

```============================================================ test session starts =============================================================
platform darwin -- Python 3.7.5, pytest-5.3.1, py-1.8.0, pluggy-0.13.1
rootdir: /Users/vagrant/Desktop/Shared/wazuh-qa/test_wazuh, inifile: pytest.ini
collected 783 items                                                                                                                          

test_basic_usage/test_basic_usage_changes.py ..                                                                                        [  0%]
test_basic_usage/test_basic_usage_create_rt_wd.py ssssssssssss                                                                         [  1%]
test_basic_usage/test_basic_usage_create_scheduled.py ............                                                                     [  3%]
test_basic_usage/test_basic_usage_delete_folder.py ..                                                                                  [  3%]
test_basic_usage/test_basic_usage_move_file.py .....                                                                                   [  4%]
test_basic_usage/test_basic_usage_rename.py ..                                                                                         [  4%]
test_basic_usage/test_basic_usage_starting_agent.py ...                                                                                [  4%]
test_checks/test_check_all.py ..........ssssssssssssssssss........                                                                     [  9%]
test_checks/test_check_others.py .........Fssssssssssssssssss........                                                                  [ 14%]
test_checks/test_checksums.py .............sssssssssssssssssssssssss............                                                       [ 20%]
test_checks/test_hard_link.py ....                                                                                                     [ 20%]
test_invalid/test_invalid.py ....                                                                                                      [ 21%]
test_recursion_level/test_recursion_level.py ......FF......FF                                                                          [ 23%]
test_moving_files/test_moving_files.py FF                                                                                              [ 23%]
test_report_changes/test_report_changes_and_diff.py ..                                                                                 [ 24%]
test_report_changes/test_report_deleted_diff.py ..                                                                                     [ 24%]
test_frequency/test_frequency.py FFFFFF                                                                                                [ 25%]
test_nodiff/test_no_diff_valid.py sssssssssssssssssss......s......ssss.sssss....s......ssss.sssss....s......ssss.sssss................ [ 37%]
s.sss..................sss..................sss....s......s.s..s.sss....s......s.s..ss.ss                                              [ 49%]
test_restrict/test_restrict_valid.py ssssssssssssssss........ssssssssssss................ssssssss....ssssssssssssssss........ssssssss. [ 61%]
...ssssssssssssssss........ssssssss....ssssssss                                                                                        [ 67%]
test_ignore/test_ignore_valid.py sssssssssssssssssss..ss....s......ssss.sssssss....s......ssss.sssssss....s......ssss.sssssss......... [ 80%]
.......s.sssss..................sssss..................sssss....s......s.s..s.sssss....s......s.s..ss.sssssssssssssssssssssssss..      [ 96%]
test_tags/test_tags.py ................                                                                                                [ 98%]
test_scan/test_scan_day.py ...                                                                                                         [ 99%]
test_scan/test_scan_day_and_time.py ..                                                                                                 [ 99%]
test_scan/test_scan_time.py ...                                                                                                        [100%]

================================================================== FAILURES ==============================================

```

- `Recursion level` tests fail at some point due to `File name too long`. Every other test fails due to timeout error.

### CODE REVIEW

File: packages/wazuh_testing/wazuh_testing/fim.py [hunk 1]
```diff
@@ -400,16 +404,15 @@ def modify_file(path, name, new_content=None, is_binary=False):
     modify_file_win_attributes(path, name)
 
 
-def change_internal_options(pattern, value, opt_path=os.path.join('/', 'var', 'ossec', 'etc', 'internal_options.conf')):
-    """Changes the value of a given parameter in linux.
+def change_internal_options(param, value, opt_path=os.path.join('/', 'var', 'ossec', 'etc', 'local_internal_options.conf')):
```
Comment (crd1985) [inline packages/wazuh_testing/wazuh_testing/fim.py]: Make function platform dependent

File: test_wazuh/test_fim/test_basic_usage/test_basic_usage_rename.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 testdir1, testdir2 = test_directories[2:]
 new_name = 'this_is_a_new_name'
 old_name = 'old_name'
-timeout = 10 if sys.platform == 'win32' else 3
+timeout = 10 if sys.platform == 'win32' or sys.platform == 'darwin' else 3
```
Comment (crd1985) [inline test_wazuh/test_fim/test_basic_usage/test_basic_usage_rename.py]: Use DEFAULT_TIMEOUT

File: packages/wazuh_testing/wazuh_testing/fim.py (hunk 1 above)
Comment (vicferpoy) [reply packages/wazuh_testing/wazuh_testing/fim.py]: Done

File: test_wazuh/test_fim/test_basic_usage/test_basic_usage_rename.py (hunk 2 above)
Comment (vicferpoy) [reply test_wazuh/test_fim/test_basic_usage/test_basic_usage_rename.py]: Done

**Pull request merged**